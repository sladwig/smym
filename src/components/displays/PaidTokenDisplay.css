.paid-token {
    font-size: 24px;
    font-weight: 500;
    border-radius: 32px;
    background-color: #f6f7fb;
    padding: 14px 19px;
}