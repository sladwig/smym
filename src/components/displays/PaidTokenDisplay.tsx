import React, { useCallback } from 'react';
import './PaidTokenDisplay.css';
import { useSpring, animated, config } from 'react-spring';
import { display } from './display';
import { CrossSvg } from '../Icon';
import { AnyCharacterT } from './types';

export const PaidTokenDisplay = ({
    word,
}: {
    word: { value: string; characters: AnyCharacterT[] };
}) => {
    const state = word.value.length;
    const { x } = useSpring({
        from: { x: 1 },
        x: state,
        config: { ...config.stiff, clamp: true },
    });
    const to = useCallback((output: number[]) => ({ range: [1, 4], output }), []);

    return (
        <animated.div
            className="paid-token"
            style={{
                opacity: x.to(to([0.3, 1])),
                borderRadius: x.to(to([0, 32])),
                fontSize: x.to(to([36, 24])),
                paddingTop: x.to(to([0, 14])),
                paddingBottom: x.to(to([0, 14])),
                paddingLeft: x.to(to([0, 19])),
                paddingRight: x.to(to([0, 19])),
            }}
        >
            <>{word.characters.map(display)}</>
            {state === 4 && (
                <span style={{ marginLeft: 15 }}>
                    <CrossSvg />
                </span>
            )}
        </animated.div>
    );
};
