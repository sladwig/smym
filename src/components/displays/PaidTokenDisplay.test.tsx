import { render } from '@testing-library/react';
import React from 'react';
import { PaidTokenDisplay } from './PaidTokenDisplay';

describe('PaidTokenDisplay', () => {
    test('renders', () => {
        const { container } = render(<PaidTokenDisplay word={{ value: '', characters: [] }} />);
        expect(container).toMatchSnapshot();
    });
});
