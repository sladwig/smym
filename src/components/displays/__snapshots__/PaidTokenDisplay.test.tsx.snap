// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaidTokenDisplay renders 1`] = `
<div>
  <div
    class="paid-token"
    style="opacity: 0.3; border-radius: 0; font-size: 36px; padding: 0px 0px 0px 0px;"
  />
</div>
`;
